core/vm/interface.go

```diff
@@ -99,13 +98,36 @@ type ChainContext interface {
 	// in the correct fork.
 	GetHeaderByNumber(uint64) *types.Header
 
-	// GetVMConfig returns the node's vm configuration
-	GetVMConfig() *Config
+	// Config returns the blockchain's chain configuration
+	Config() *params.ChainConfig
+}
 
-	CurrentHeader() *types.Header
+type SystemEVM interface {
```

Comment:
I don't think this makes sense anymore because this is called 'SystemEVM' but it no longer hard codes the system address as the caller.

I see 2 options:
* Call it "SystemEVM" and always hardcode the caller to be the zero address and the gas price to zero, this means caller can be removed as a parameter to all of the methods.
* Call it "EVMCaller" and let both caller and gas price be provided as parameters, this makes it a more general use object that we can use for applying transactions as well as making system calls. I prefer this route because it should allow us to always call the evm in the same way.

Reply:
Sounds good :+1: